### PR TITLE
HCAP-389: Single site POST - BE

### PR DESCRIPTION
**Pull request by fwkendall**
- Discovered nothing was using the existing sites POST endpoint - all our scripts to create sites go directly via the DB using saveSites in employers.js
- Repurposed the existing endpoint to be a single site POST endpoint, limited to MOH and with new validation
- The main validation difference between POST/PATCH is that single site POST requires Site ID, and also supports Operator Name - these are both already supported in the batch DB operation (which has its own validation as well)

Incidentally this is how I'm noticing we forgot to put the Operator Name in the single site view FE. 🙃 

Other cleanup:
- Split the employer form (EEOI) tests vs. site tests into separate files
- Added new tests for single site POST

### CODE REVIEW

File: server/tests/employer.site.test.js [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+    operatorContactLastName: 'CCC',
+    operatorEmail: 'test@hcpa.fresh',
+    operatorPhone: '2219909090',
+    siteContactFirstName: 'NNN',
+    siteContactLastName: 'PCP',
+    siteContactPhoneNumber: '2219909091',
+    siteContactEmailAddress: 'test.site@hcpa.fresh',
+  };
+
+  it('Create new single site, receive success', async () => {
```
Comment (fwkendall) [inline server/tests/employer.site.test.js:79]: This test and the one below are new, everything else is copied from employer form tests (should be separate commits for easy review)

File: server/tests/employer.form.test.js [hunk 2]
```diff
@@ -214,142 +155,19 @@ describe('Server V1 Form Endpoints', () => {
...
-  it('fetches the list of employers', async () => {
+  it('fetches the list of employer expression of interest forms', async () => {
     const res = await getEmployers({
       isSuperUser: false, isMoH: false, isHA: true, regions: ['Vancouver Island'],
     });
     expect(res).toEqual([]);
   });
 
-  it('fetches a single employer', async () => {
+  it('fetches a single employer expression of interest form', async () => {
```
Comment (fwkendall) [inline server/tests/employer.form.test.js:165]: Only "real" change in this file is updating the descriptions on lines 158 & 165 - the rest has just been moved to the sites test file

File: server/services/employers.js [hunk 3]
```diff
@@ -11,6 +11,23 @@ const getEmployers = async (user) => {
 
 const getEmployerByID = async (id) => dbClient.db[collections.EMPLOYER_FORMS].findDoc({ id });
 
+const saveSingleSite = async (siteJson) => {
+  const res = await dbClient.db.saveDoc(collections.EMPLOYER_SITES, siteJson);
+  const { siteId } = res;
+  switch (res.status) {
```
Comment (fwkendall) [inline server/services/employers.js:17]: this is where things are falling over, but I can't debug locally - might need to change the way I'm attempting to save the record on line 15. I do want to be able to distinguish duplicates from other failure reasons and then have server.js return error codes the FE can use. I'm think we'll want to be able to let the user know while they're still in the modal (so if it's a dupe, we can keep the modal open with a failure for the dupe in case they just made a typo)

File: server/tests/employer.site.test.js [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+    emailAddress: 'freddy@example.com',
+    interested: 'yes',
+    nonHCAP: 'yes',
+    crcClear: 'yes',
+    preferredLocation: 'Fraser',
+  };
+
+  // Used for single site POST
+  // siteId must be assigned by the test
+  const siteBaseFields = {
```
Comment (fwkendall) [inline server/tests/employer.site.test.js:61]: this is new to support single site POST

File: server/tests/employer.site.test.js [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+  it('Get sites, receive all successfully', async () => {
+    const res = await getSites();
+    expect(res).toEqual(
+      expect.arrayContaining(
+        [site].map((item) => (expect.objectContaining(item))),
+      ),
+    );
+  });
+
+  it('Gets a single site', async () => {
```
Comment (fwkendall) [inline server/tests/employer.site.test.js:138]: Updated this test so it creates the site it's loading, instead of assuming which site has which ID

File: server/services/employers.js [hunk 6]
```diff
@@ -11,6 +11,11 @@ const getEmployers = async (user) => {
 
 const getEmployerByID = async (id) => dbClient.db[collections.EMPLOYER_FORMS].findDoc({ id });
 
+const saveSingleSite = async (siteJson) => {
+  const res = await dbClient.db.saveDoc(collections.EMPLOYER_SITES, siteJson);
```
Comment (Zenen-freshworks) [inline server/services/employers.js:15]: Woah, cool! I wasn't aware of this syntax.

**Review by Zenen-freshworks — APPROVED** [review]
Looks good! I really appreciate the work put into refactoring some of the tests to make them less interdependent.

**Review by mauriciofreshw — APPROVED** [review]
